recipes/openvdb/all/conanfile.py

```diff
@@ -16,232 +21,283 @@ class OpenVDBConan(ConanFile):
-            "Visual Studio": "15",  # Should we check toolset?
-            "gcc": "6.3.1",
-            "clang": "3.8",
-            "apple-clang": "3.8",
-            "intel": "17",
-        }
-
-    def export_sources(self):
-        self.copy("CMakeLists.txt")
-        for patch in self.conan_data.get("patches", {}).get(self.version, []):
-            self.copy(patch["patch_file"])
+        if Version(self.version) >= "10.0.0":
+            # https://github.com/AcademySoftwareFoundation/openvdb/blob/v10.0.1/doc/dependencies.txt#L56-L84
+            return {
+                "msvc": "192.8",
+                "Visual Studio": "16",
+                "gcc": "9.3.1",
+                "clang": "5.0",
+                "apple-clang": "12.0",
+                "intel-cc": "19",
+            }
+        else:
+            # https://github.com/AcademySoftwareFoundation/openvdb/blob/v9.1.0/doc/dependencies.txt#L56-L84
+            return {
+                "msvc": "191.0",
+                "Visual Studio": "15",
+                "gcc": "6.3.1",
+                "clang": "3.8",
+                "apple-clang": "10.0",
+                "intel-cc": "17",
+            }
 
     def config_options(self):
         if self.settings.os == "Windows":
             del self.options.fPIC
+        if is_msvc(self):
+            # Supported by GCC and Clang only
+            del self.options.use_colored_output
+        if Version(self.version) < "10.0.0":
+            del self.options.use_explicit_instantiation
+            del self.options.use_delayed_loading
 
     def configure(self):
         if self.options.shared:
-            del self.options.fPIC
+            self.options.rm_safe("fPIC")
+
+    def layout(self):
+        cmake_layout(self, src_folder="src")
+
+    def package_id(self):
+        # with_exr is deprecated and has no effect
+        del self.info.options.with_exr
 
     def requirements(self):
-        self.requires("boost/1.79.0")
-        self.requires("onetbb/2020.3")
-        self.requires("openexr/2.5.7")  # required for IlmBase::Half
+        # https://github.com/AcademySoftwareFoundation/openvdb/blob/v10.0.1/doc/dependencies.txt#L36-L84
+        self.requires("boost/1.83.0", transitive_headers=True)
+        self.requires("onetbb/2021.10.0", transitive_headers=True, transitive_libs=True)
+        if self.options.use_imath_half:
+            self.requires("imath/3.1.9", transitive_headers=True, transitive_libs=True)
         if self.options.with_zlib:
             self.requires("zlib/[>=1.2.11 <2]")
-        if self.options.with_exr:
-            # Not necessary now. Required for IlmBase::IlmImf
-            self.requires("openexr/2.5.7")
         if self.options.with_blosc:
-            self.requires("c-blosc/1.21.1")
+            self.requires("c-blosc/1.21.5")
         if self.options.with_log4cplus:
-            self.requires("log4cplus/2.0.7")
+            # TODO: OpenVDB requires log4cplus v1.*, which is not available in CCI
```

Comment:
What does this one mean? 1.2.2 does exist, so what's the comment referring to?

Reply:
Yeah, I added the v1.2.2 in a later PR (#21904). :)
I'll update the recipe.